src/metadata/metadata.c

```diff
@@ -27,7 +27,10 @@
 #include "dislocker/metadata/metadata_config.h"
 #include "dislocker/metadata/print_metadata.h"
 #include "dislocker/dislocker.priv.h"
-#include <sys/mount.h>
+
+#include <sys/ioctl.h>
+
+#define BLKSSZGET  _IO(0x12,104)/* get block device sector size */
```

Comment:
How will it behave when the file you're reading isn't a partition but a dumped, plain file?

Reply:
If dis_meta->volume_header->sector_size has a value, ioctl will not be executed.
If file cannot get sector size via ioctl, then the sector size will be set as 512 as default. It seems that it has to be handled in this way, because in Windows 10 1903 BitLocker of exFAT file system cannot get sector size from DBR.